src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1148,14 +1156,34 @@ public new Task<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior, C
         internal ManualResetValueTaskSource<NpgsqlConnector> ExecutionCompletion { get; }
             = new();
 
+        NpgsqlConnector? TryGetBoundConnector(NpgsqlConnection? connection)
```

Comment:
Naming nit: TryGet in .NET usually returns a bool and has an out var (e.g. Dictionary.TryGetValue). This could just be called GetBoundConnector.